src/EnergyPlus/GroundTemperatureModeling/GroundTemperatureModelManager.cc

```diff
@@ -81,27 +81,33 @@ namespace GroundTemperatureManager {
 
         // Locals
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int objectType(0);
+        GroundTempObjType objectType(GroundTempObjType::Unassigned);
 
         auto &CurrentModuleObjects = state.dataGrndTempModelMgr->CurrentModuleObjects;
 
         std::string objectType_str_UPPERCase = UtilityRoutines::MakeUPPERCase(objectType_str);
 
         // Set object type
-        if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_KusudaGroundTemp))) {
-            objectType = objectType_KusudaGroundTemp;
-        } else if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_FiniteDiffGroundTemp))) {
-            objectType = objectType_FiniteDiffGroundTemp;
-        } else if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_SiteBuildingSurfaceGroundTemp))) {
-            objectType = objectType_SiteBuildingSurfaceGroundTemp;
-        } else if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_SiteShallowGroundTemp))) {
-            objectType = objectType_SiteShallowGroundTemp;
-        } else if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_SiteDeepGroundTemp))) {
-            objectType = objectType_SiteDeepGroundTemp;
-        } else if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_SiteFCFactorMethodGroundTemp))) {
-            objectType = objectType_SiteFCFactorMethodGroundTemp;
-        } else if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(objectType_XingGroundTemp))) {
-            objectType = objectType_XingGroundTemp;
+        if (objectType_str_UPPERCase == UtilityRoutines::MakeUPPERCase(CurrentModuleObjects(static_cast<int>(GroundTempObjType::KusudaGroundTemp)))) {
+            objectType = GroundTempObjType::KusudaGroundTemp;
```

Comment:
Probably can eliminate GroundTemp from the enumerated value itself since it is part of the enumeration class name.

Reply:
One thing to note is that most of these ground temp files are hanging out on the EnergyPlus namespace. That's probably something we should correct at some point, but that may affect how we handle your comment here.